digiroad2-api-oth/src/main/scala/fi/liikennevirasto/digiroad2/Digiroad2Api.scala

```diff
@@ -1082,8 +1092,12 @@ class Digiroad2Api(val roadLinkService: RoadLinkService,
     val newLinearAssets = extractNewLinearAssets(typeId, parsedBody \ "newLimits")
     val existingAssets = usedService.getPersistedAssetsByIds(typeId, existingAssetIds)
 
+    val asset = if (newLinearAssets.nonEmpty) newLinearAssets.head
+                else createFakeNewLinearAssetForValidations (existingAssets.head ,valueOption.get )
+
```

Comment:
Check this because I think the extractNewLinearAssets method, who returns newLinearAssets values always return a Seq[NewLinearAsset], ok can be empty but in that case, may don't validate anything and leaves the process continues, instead of force other values only to pass.

Reply:
valueOption.get can give an exception, verify this situation

Reply:
I believe that ignoring the validation for existing assets is not right even though we are force transform from one object to another but with that in mind and the valueOption/Seq[NewLinearAsset] I think the code should look better this way:

```scala
  def createFakeNewLinearAssetsForValidations(existingAssets: Seq[PersistedLinearAsset], inputValues: Option[Value]): Seq[NewLinearAsset] = {
    inputValues match {
      case Some(values) => existingAssets.map(existingAsset => NewLinearAsset(existingAsset.linkId, existingAsset.startMeasure,
        existingAsset.endMeasure, values, existingAsset.sideCode, existingAsset.vvhTimeStamp, existingAsset.geomModifiedDate))
      case _ => Seq()
    }
  }

    val assets = newLinearAssets ++ createFakeNewLinearAssetsForValidations(existingAssets, valueOption)
    assets.foreach(asset => usedService.validateCondition(asset))
```